src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs

```diff
@@ -7,108 +7,77 @@
-
-        /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Func<CookieValidatePrincipalContext, Task> OnValidatePrincipal { get; set; }
+        public Func<CookieValidatePrincipalContext, Task> OnValidatePrincipal { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignInContext> OnResponseSignIn { get; set; }
+        public Action<CookieResponseSignInContext> OnResponseSignIn { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignedInContext> OnResponseSignedIn { get; set; }
+        public Action<CookieResponseSignedInContext> OnResponseSignedIn { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieResponseSignOutContext> OnResponseSignOut { get; set; }
+        public Action<CookieResponseSignOutContext> OnResponseSignOut { get; set; } = context => { };
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieApplyRedirectContext> OnApplyRedirect { get; set; }
+        public Action<CookieApplyRedirectContext> OnApplyRedirect { get; set; } = context => context.Response.Redirect(context.RedirectUri);
 
         /// <summary>
         /// A delegate assigned to this property will be invoked when the related method is called
         /// </summary>
-        public Action<CookieExceptionContext> OnException { get; set; }
+        public Action<CookieExceptionContext> OnException { get; set; } = context => { };
 
         /// <summary>
         /// Implements the interface method by invoking the related delegate method
         /// </summary>
         /// <param name="context"></param>
         /// <returns></returns>
-        public virtual Task ValidatePrincipal(CookieValidatePrincipalContext context)
-        {
-            return OnValidatePrincipal.Invoke(context);
-        }
+        public virtual Task ValidatePrincipal(CookieValidatePrincipalContext context) => OnValidatePrincipal(context);
 
         /// <summary>
         /// Implements the interface method by invoking the related delegate method
         /// </summary>
         /// <param name="context"></param>
-        public virtual void ResponseSignIn(CookieResponseSignInContext context)
-        {
-            OnResponseSignIn.Invoke(context);
-        }
+        public virtual void ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
```

Comment:
Can you remind me why we decided on method + delegate again?  What does the method allow that the public delegate property doesn't?